cmd/pctl/main.go

```diff
@@ -84,6 +87,79 @@ func showCmd() *cli.Command {
 	}
 }
 
+func installCmd() *cli.Command {
+	return &cli.Command{
+		Name:      "install",
+		Usage:     "generate a profile subscription for a profile in a catalog",
+		UsageText: "pctl --catalog-url <URL> install --subscription-name pctl-profile --namespace default --branch main --config-secret configmap-name --out profile_subscription.yaml <CATALOG>/<PROFILE>",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:        "subscription-name",
+				DefaultText: "pctl-profile",
+				Value:       "pctl-profile",
+				Usage:       "The name of the subscription.",
+			},
+			&cli.StringFlag{
+				Name:        "namespace",
+				DefaultText: "default",
+				Value:       "default",
+				Usage:       "The namespace to use for generating resources.",
+			},
+			&cli.StringFlag{
+				Name:        "branch",
+				Value:       "main",
+				DefaultText: "main",
+				Usage:       "The branch to use on the repository in which the profile is.",
+			},
+			&cli.StringFlag{
+				Name:  "config-secret",
+				Value: "",
+				Usage: "The name of the ConfigMap which contains values for this profile.",
+			},
+			&cli.StringFlag{
+				Name:        "out",
+				Value:       "profile_subscription.yaml",
+				DefaultText: "profile_subscription.yaml",
+				Usage:       "Filename to use for the generated content.",
+			},
+		},
+		Action: func(c *cli.Context) error {
+			profilePath, catalogURL, err := parseArgs(c)
+			if err != nil {
+				_ = cli.ShowCommandHelp(c, "install")
+				return err
+			}
+
+			branch := c.String("branch")
+			subName := c.String("subscription-name")
+			namespace := c.String("namespace")
+			configValues := c.String("config-secret")
+			filename := c.String("out")
+
+			parts := strings.Split(profilePath, "/")
+			if len(parts) < 2 {
+				_ = cli.ShowCommandHelp(c, "install")
+				return errors.New("both catalog name and profile name must be provided")
+			}
+			catalogName, profileName := parts[0], parts[1]
+
+			fmt.Printf("generating subscription for profile %s/%s:\n\n", catalogName, profileName)
+			var w writer.Writer = &writer.FileWriter{Filename: filename}
```

Comment:
```suggestion
			w := &writer.FileWriter{Filename: filename}
```
for continuity?

suppose you could also inline this on 156, but that is a personal style choice so have no strong feelings

Reply:
Ah yes, I used the interface definition there before because of the possible overwrite after. This was previously an stdout write which was optionally overwritten but the variable had to stay an interface type. :)

Now, that's not needed, so yeah, I can remove it. :)